src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -418,15 +431,33 @@ private async Task<JObject> GetLocalSettingsJsonAsJObjectAsync()
         public override async Task RunAsync()
         {
             await PreRunConditions();
+            var isVerbose = VerboseLogging.HasValue && VerboseLogging.Value;
 
             var isCurrentProcessNet6Build = RuntimeInformation.FrameworkDescription.Contains(Net6FrameworkDescriptionPrefix);
-            if (isCurrentProcessNet6Build && ShouldLaunchInProcNet8AsChildProcess() && await IsInProcNet8Enabled())
+            // If --runtime param is set, handle runtime param logic; otherwise we infer the host to launch on startup
+            if (SetHostRuntime != null)
             {
-                await StartInProc8AsChildProcessAsync();
-                return;
+                // Check if we should start child process from user specified host runtime and return
+                var shouldStartChildProcess = await ShouldStartChildProcessFromHostRuntime(isCurrentProcessNet6Build, isVerbose);
+                var isOutOfProcSpecified = IsOutOfProcSpecifiedFromHostRuntime();
```

Comment:
This isn't used outside of the if statement so should be defined right before calling await StartHostAsChildProcessAsync